src/coreclr/jit/utils.cpp

```diff
@@ -2211,6 +2211,126 @@ bool FloatingPointUtils::hasPreciseReciprocal(float x)
     return mantissa == 0 && exponent != 0 && exponent != 127;
 }
 
+static bool isNegative(float val)
+{
+    return *reinterpret_cast<INT32*>(&val) < 0;
+}
+
+static bool isNegative(double val)
+{
+    return *reinterpret_cast<INT64*>(&val) < 0;
+}
+
+static bool isNaN(float val)
+{
+    return val != val;
+}
+
+static bool isNaN(double val)
+{
+    return val != val;
+}
+
+//------------------------------------------------------------------------
+// max_double: This matches the IEEE 754:2019 `maximum` function
+//    It propagates NaN inputs back to the caller and
+//    otherwise returns the larger of the inputs. It
+//    treats +0 as larger than -0 as per the specification.
+//
+// Arguments:
+//    val1 - left operand
+//    val2 - right operand
+//
+// Return Value:
+//    Either val1 or val2
+//
+
+double FloatingPointUtils::max_double(double val1, double val2)
```

Comment:
nit: The convention so far is that this would be named just `max` since it can differentiate based on overload

`_double` is only used for cases like `infinte_double` where there are no parameters and we need to overload

Reply:
I renamed to `maximum/minimum` to match IEEE754 name and avoid clash with max/min macros